### ISSUE TITLE
Unsupported Database: PostgreSQL 16.1 when upgrade to 10.4.1

### ISSUE THREAD

**Issue by rakum23** [flyway/flyway#3819]
###### Which version and edition of Flyway are you using?
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name  flywayInitializer defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Unsupported Database: PostgreSQL 16.1

**Comment by Barry-RG** [comment]
@rakum23 
Which version of Flyway are you using and have you included `flyway-database-postgresql` as a dependency in your project?

**Comment by rakum23** [comment]
we are using flyway 9.22.3 with configuration like: 

  flyway:
    locations: classpath:/db/migration
    database-name: postgres
    enabled: true

it is working perfectly with this and when upgrading to flyway 10.4.1 version the issue happens. The Spring boot version is 3.1.6

**Comment by rakum23** [comment]
https://github.com/flyway/flyway/issues/3780

It has come to our attention that the move to a more modular Flyway in version V10.0.0 may not have been made clear enough in our [release notes](https://documentation.red-gate.com/flyway/release-notes-and-older-versions/release-notes-for-flyway-engine).

As we attempt to move towards a much wider range of supported databases and database types we have found ourselves needing to extract database support out from flyway-core and into their own modules in order to improve maintainability and distribution size.

As a result, databases, with the exception of in-memory or file databases such as H2 or SQLite, have been moved into their own packages which need to be added to your project when upgrading to Flyway V10.0.0. This includes whether you are using Flyway as a dependency in your project or using the Gradle or Maven plugins.
